Adicionando tabelas e atributos

savioc2 opened the pull request:
PedroMoraes39 <phcmoraes39@gmail.com>

savioc2 committed:
adicionando tabelas e atributos

PedroMoraes39 <phcmoraes39@gmail.com>

fmaachadoo commented on database/ddl.sql:
```diff
@@ -1,15 +1,84 @@
+    CONSTRAINT fk_Produto FOREIGN KEY (id_Produto) REFERENCES Produto(id)
+);
+
+CREATE TABLE IF NOT EXISTS Itens_da_compra (
+    id INT,
+    id_Produto INT,
+    PRIMARY KEY(id),
+    CONSTRAINT fk_produto FOREIGN KEY(id_Produto) REFERENCES Produto(id)
+);
```
Acredito que o nome dessa tabela poderia melhorar

fmaachadoo replied:
Qual o sentido dessa tabela?

savioc2 replied:
Essa tabela foi revisada e ela será removida

fmaachadoo commented on database/ddl.sql:
```diff
@@ -1,15 +1,84 @@
+
+CREATE TABLE IF NOT EXISTS Carrinho (
+    id INT,
+    id_Inventario INT,
```
Pq o carrinho tem um `id_Inventario`?

savioc2 replied:
Realmente, era pra ser Id do item, para saber qual produto estar no carrinho.

fmaachadoo commented on database/ddl.sql:
```diff
@@ -1,15 +1,84 @@
+
+CREATE TABLE IF NOT EXISTS Pagamento(
+    id INT,
+    id_Itens INT,
```
pq pagamento possui `id_itens`?

savioc2 replied:
Porque durante o pagamento, é necessário saber qual item foi comprado, para poder gerar a nota.

fmaachadoo reviewed (changes requested):
Acredito que podemos revisar as estruturas desse banco de dados de uma forma melhor se também tivermos scripts para inserção de dados no banco.

PedroMoraes39 committed:
Modificando tabelas desnecessarias

savioc2 committed:
adicionando dml

savioc2 committed:
adicionando models

Co-authored-by: samuelcastro7 <SamuelCastro2010@gmail.com>

fmaachadoo commented on grabber_backend/database_controller/databaseHandler.py:
Nomes de arquivos python devem ser snake_case.

fmaachadoo commented on grabber_backend/database_controller/models/carrinho.py:
O nosso padrão de código é em inglês.
Nomes de arquivos, variáveis e funções devem ser em inglês.

fmaachadoo commented on grabber_backend/database_controller/models/models.py:
Lembrar de deletar esse arquivo que está todo comentado

fmaachadoo commented on grabber_backend/tests/test_database_models.py:
```diff
@@ -0,0 +1,260 @@
+
+    #     # Verificar se o usuário foi inserido corretamente
+    #     usuario_db = self.session.query(Usuario).filter_by(cpf=123456789).first()
+    #     self.assertEqual(usuario_db.nome, 'John Doe')
+
+    #     # Atualizar o nome do usuário
+    #     usuario_db.nome = 'Jane Doe'
+    #     self.session.commit()
+
+    #     # Verificar se a atualização foi bem-sucedida
+    #     usuario_atualizado = self.session.query(Usuario).filter_by(cpf=123456789).first()
+    #     self.assertEqual(usuario_atualizado.nome, 'Jane Doe')
+
+    #     # Remover o usuário
+    #     self.session.delete(usuario_atualizado)
+    #     self.session.commit()
+
+    #     # Verificar se o usuário foi removido corretamente
+    #     usuario_removido = self.session.query(Usuario).filter_by(cpf=123456789).first()
+    #     self.assertIsNone(usuario_removido)
+
+    # def test_carrinho(self):
+    #     # Casos de teste para a classe Carrinho
+    #     # ...
+
+    # def test_pagamento(self):
+    #     # Casos de teste para a classe Pagamento
+    #     # ...
+
+    # def test_pedidos(self):
+    #     # Casos de teste para a classe Pedidos
+    #     # ...
+
+    # def test_estoquista(self):
+    #     # Casos de teste para a classe Estoquista
+    #     # ...
+
+    # def test_consumidor(self):
+    #     # Casos de teste para a classe Consumidor
+    #     # ...
```
Qual o sentido de comentar todas as funções? Não seria melhor deletar?

fmaachadoo commented on grabber_backend/tests/test_database_models.py:
```diff
@@ -0,0 +1,260 @@
+#         stockist_consulted = self.session.query(estoquista).filter_by(id=1).first()
+
+#         # Verificar se os valores são os mesmos
+#         self.assertEqual(stockist_consulted.controleEstoque, 'Controle de Estoque')
+
+#     def test_pedidos(self):
+#         # Criar um objeto de Pedidos
+#         requests = pedidos(id=1, id_Pagamento=1, id_Carrinho=1, id_Usuario=1234567890, data_pedido=20220601, tipo='Normal')
+
+#         # Adicionar o objeto à sessão
+#         self.session.add(requests)
+#         self.session.commit()
+
+#         # Consultar o objeto da tabela Pedidos
+#         requests_consulted = self.session.query(pedidos).filter_by(id=1).first()
+
+#         # Verificar se os valores são os mesmos
+#         self.assertEqual(requests_consulted.id_Pagamento, 1)
+#         self.assertEqual(requests_consulted.id_Carrinho, 1)
+#         self.assertEqual(requests_consulted.id_Usuario, 1234567890)
+#         self.assertEqual(requests_consulted.data_pedido, 20220601)
+#         self.assertEqual(requests_consulted.tipo, 'Normal')
+
+#     def test_consumidor(self):
+#         # Criar um objeto de Consumidor
+#         consumer = consumidor(id=1, id_Carrinho=1, id_Pagamento=1)
+
+#         # Adicionar o objeto à sessão
+#         self.session.add(consumer)
+#         self.session.commit()
+
+#         # Consultar o objeto da tabela Consumidor
+#         consumer_consulted = self.session.query(consumidor).filter_by(id=1).first()
+
+#         # Verificar se os valores são os mesmos
+#         self.assertEqual(consumer_consulted.id_Carrinho, 1)
+#         self.assertEqual(consumer_consulted.id_Pagamento, 1)
+
+# if __name__ == '__main__':
+#     unittest.main()
```
?

fmaachadoo committed:
Merge branch 'main' of https://github.com/UNB-PI2-Grupo3-BracoRobotico/grabber-backend into feat/modelagemdb

fmaachadoo committed:
chore: delete unused file

fmaachadoo committed:
chore: delete pytest comments

fmaachadoo committed:
chore: rename file

fmaachadoo committed:
chore: delete comments

fmaachadoo committed:
fix: cpf must be bigint with 11 digits

fmaachadoo committed:
fix: localizacao must be integers

SamuelCastro7 committed:
Classes Models em inglês

Alteração das classes dos models para inglês.

SamuelCastro7 committed:
Tentativa para teste da model

Tentativa de uma maneira diferente de implementar uma classe para ver se vai passar ou não nos testes.

fmaachadoo commented:
PR #44 Já resolveu isso
